### PR TITLE
Small adjustment in CLI 

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

#397 
#398

# Overview

- [ ] I have fixed inconsistencies in CLI addressed by #397 and #398 and adjusted integration tests to make them compliant with new behavior

# Comments

### CODE REVIEW

File: exe/wallet/Main.hs [hunk 1]
```diff
@@ -311,15 +314,19 @@ exec execServer manager args
...
         res <- runClientM cmd env
         case res of
             Right a -> do
-                TIO.hPutStrLn stderr "Ok."
-                BL8.putStrLn (encode a)
+                if silentMode then
+                    TIO.hPutStr stderr ""
+                else do
+                    TIO.hPutStrLn stderr "Ok."
+                    BL8.putStrLn (encode a)
```
Comment (KtorZ) [inline exe/wallet/Main.hs]: What's the rational / requirement for the `silentMode` :thinking: ?
Also, avoid `Bool` where you can and favor a concrete data-type that conveys the semantic better:

```hs
data Mode = Silent | Verbose
```

Although here, I don't think we need this ?

Comment (KtorZ) [reply exe/wallet/Main.hs]: Instead, I'll suggest to:

- split `runClient` into two smaller functions:
    - `sendRequest`
    - `handleResponse`

This way, you may re-use `sendRequest` easily to check that a wallet exists without doing all the standard handling.

File: exe/wallet/Main.hs [hunk 2]
```diff
@@ -313,9 +318,15 @@ exec execServer manager args
         -> ClientM a
         -> IO ()
     runClient encode cmd = do
+        res <- sendRequest cmd
+        handleResponse encode res
+
+    sendRequest cmd = do
```
Comment (piotr-iohk) [inline exe/wallet/Main.hs:328]: @KtorZ I did what you suggested, but failed with providing function declaration for both `sendRequest` and `handleResponse`, tried this, but it didn't work:

```haskell
    handleResponse
        :: forall a. ()
        => (a -> BL.ByteString)
        -> IO (Either ServantError a)
        -> IO ()


    sendRequest
        :: forall a. ()
        => ClientM a
        -> IO (Either ServantError a)
```

Comment (piotr-iohk) [reply exe/wallet/Main.hs:328]: ok, got it... :man_facepalming:

File: exe/wallet/Main.hs [hunk 3]
```diff
@@ -245,24 +246,28 @@ exec execServer manager args
...
       args `isPresent` command "create" = do
         wId <- args `parseArg` argument "wallet-id"
         ts <- args `parseAllArgs` longOption "payment"
-        wPwd <- getPassphrase
-        runClient Aeson.encodePretty $ createTransaction (ApiT wId) $
-            PostTransactionData
-                ts
-                (ApiT wPwd)
+        res <- sendRequest $ getWallet $ ApiT wId
+        if (isRight res) then do
```
Comment (KtorZ) [inline exe/wallet/Main.hs:256]: Just a minor thing but pattern-matching `case res of Right _; Left _;` is more idiomatic in Haskell than `if isXXX then else`

**Review by KtorZ — APPROVED** [review]
:ok_hand:

**Pull request merged**